PhpAmqpLib/Channel/AbstractChannel.php

```diff
@@ -283,8 +288,8 @@ protected function build_msg_body($body_size) {
 
             $body_received = bcadd($body_received, mb_strlen($payload, 'ASCII'), 0);
 
-            if ( ! is_null($this->body_size_max) && $body_received > $this->body_size_max ) {
-                $msg->is_truncated = true;
+            if (!is_null($this->body_size_max) && $body_received > $this->body_size_max ) {
+                $msg->is_truncated = true; // $msg?
```

Comment:
Actually this is a bug when I introduced this function here: https://github.com/eexit/php-amqplib/blob/minor-refacto/PhpAmqpLib/Channel/AbstractChannel.php#L266

I think build_msg_body should return the new body inside an array which also includes a boolean to see if message is truncated or not.